init/gneiss-broker-message.adb

```diff
@@ -249,6 +214,32 @@ is
       end if;
    end Process_Message_Request;
 
+   procedure Process_Memory_Request (Fds_In  :        Gneiss_Syscall.Fd_Array;
+                                     Fds_Out :    out Gneiss_Syscall.Fd_Array;
+                                     Valid   :    out Boolean)
+   is
+      Success : Integer;
+   begin
+      Valid   := False;
+      Fds_Out := (others => -1);
+      if Fds_In (Fds_In'First) < 0 then
+         return;
+      end if;
+      Fds_Out (Fds_Out'First + 2) := Fds_In (Fds_In'First);
+      Gneiss_Syscall.Memfd_Seal (Fds_Out (Fds_Out'First + 2), Success);
```

Comment:
According to the man page this will return an error:
> Currently, file seals can be applied only to a file descriptor returned by memfd_create(2) (if the MFD_ALLOW_SEALING was employed).  On other filesystems, all fcntl() operations that operate on seals will return EINVAL.

If the sealing returns an error the broker will answer with a reject message.